commit 86618ea73c942d90e468d4250a24a8178f903b26
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2019-07-08T16:10:09+02:00 (3 months ago)
    
    TRITON-1782 The extra quota attribute should not be displayed for Bhyve VMs

diff --git a/lib/common/vm-common.js b/lib/common/vm-common.js
index 5453f01..e095ebf 100644
--- a/lib/common/vm-common.js
+++ b/lib/common/vm-common.js
@@ -192,6 +192,9 @@ function translateVm(obj, fullObject) {
         if (obj.brand === 'bhyve' && obj.free_space !== undefined) {
             vm.free_space = obj.free_space;
         }
+        if (obj.brand === 'bhyve' && obj.flexible_disk_size) {
+            delete vm.quota;
+        }
     } else {
         vm.image_uuid = obj.image_uuid;
     }
diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 0cf5fc4..83179bd 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -300,6 +300,13 @@ function listVms(req, res, next) {
                     return done(err);
                 }
 
+                for (var i = 0; i < vms.length; i += 1) {
+                    if (vms[i].brand && vms[i].brand === 'bhyve' &&
+                        vms[i].flexible_disk_size) {
+                        delete vms[i].quota;
+                    }
+                }
+
                 req.vms = vms;
 
                 return done();
@@ -328,7 +335,7 @@ function listVms(req, res, next) {
 function getVmProc(req, res, next) {
     req.log.trace({ vm_uuid: req.params.uuid }, 'GetVmProc start');
     var error;
-    var m = req.vm;
+var m = req.vm;
 
     // Skip calling CNAPI when a VM hasn't been allocated to a server
     if (m.server_uuid === undefined) {
@@ -2659,6 +2666,12 @@ function _loadVms(req, res, next) {
         }
 
         if (vms) {
+            for (var i = 0; i < vms.length; i += 1) {
+                if (vms[i].brand && vms[i].brand === 'bhyve' &&
+                    vms[i].flexible_disk_size) {
+                    delete vms[i].quota;
+                }
+            }
             req.vms = vms;
         }
 
diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index 8a53444..3125346 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -5,11 +5,12 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
- * A brief overview of this source file: what is its purpose.
+ * This is the VM provision workflow, used to create provision jobs
+ * for Triton VMs.
  */
 
 var async = require('async');
@@ -23,7 +24,7 @@ var nfsVolumes = require('./nfs-volumes');
 
 var wfapiUrl;
 
-var VERSION = '8.2.1';
+var VERSION = '8.2.2';
 
 
 /*
@@ -231,6 +232,10 @@ function preparePayload(job, cb) {
                 payload[field] = job.params.image[field];
             }
         });
+
+        // Rely into default vmadm values with `disks` and `flexible_disk_size`
+        // for KVM/Bhyve VMs:
+        delete payload.quota;
     } else {
         payload['image_uuid'] = params['image_uuid'];
 
diff --git a/test/vms.disks.test.js b/test/vms.disks.test.js
index d085daa..06b1e49 100644
--- a/test/vms.disks.test.js
+++ b/test/vms.disks.test.js
@@ -295,6 +295,8 @@ exports.check_added_disk = function check_added_disk(t) {
             t.done();
             return;
         }
+        // Verify that there is no 'quota' attribute on this case:
+        t.ok(!vm.quota, 'No VM.quota with flexible_disk');
 
         var disks = vm.disks;
         t.equal(disks.length, 2);
